clang/build/build-cppx.sh

```diff
@@ -24,40 +24,18 @@ STAGING_DIR=$(pwd)/staging
 rm -rf ${STAGING_DIR}
 mkdir -p ${STAGING_DIR}
 
-# Setup llvm root directory
-git clone --single-branch https://github.com/llvm-mirror/llvm.git
-
-# Checkout & configure clang
-pushd llvm/tools
-git clone --depth 1 --single-branch -b compiler-explorer https://gitlab.com/lock3/clang.git
-
-# Load llvm revisions for this compiler build
-pushd clang
-source compiler-explorer-llvm-commits.sh
-popd
-
-popd
-
-# Adjust llvm revision to match the cppx compiler's specified revision
-(cd llvm && git reset --hard ${CE_LLVM_COMMIT})
-
-# Checkout and configure libcxx
-pushd llvm/projects
-git clone --single-branch https://github.com/llvm-mirror/libcxx.git
-(cd libcxx && git reset --hard ${CE_LIBCXX_COMMIT})
-# Hack for new glibc not containing xlocale.h
-perl -pi -e 's/defined\(__GLIBC__\) \|\| defined\(__APPLE__\)/defined(__APPLE__)/' libcxx/include/__locale
-git clone --single-branch https://github.com/llvm-mirror/libcxxabi.git
-(cd libcxxabi && git reset --hard ${CE_LIBCXXABI_COMMIT})
-popd
+# Setup llvm-project checkout
+git clone --depth 1 --branch compiler-explorer https://gitlab.com/lock3/clang.git llvm-project
 
+# Setup build directory and build configuration
 mkdir build
 cd build
-cmake -G "Unix Makefiles" ../llvm \
+cmake -DLLVM_ENABLE_PROJECTS=clang -G "Unix Makefiles" ../llvm-project/llvm \
```

Comment:
Useful to know: are the any other projects we might need to enable in our other builds when we move too? like asan/libc++ or similar?

Reply:
As it turns out, yes `clang;libcxx;libcxxabi` is the correct configuration, this seems to have resulted in `cassert` not being found.